MarathonRecomp/api/Sonicteam/SoX/Engine/Doc.h

```diff
@@ -12,11 +15,13 @@ namespace Sonicteam::SoX::Engine
         xpointer<void> m_pVftable;
         MARATHON_INSERT_PADDING(4);
         xpointer<DocMode> m_pDocMode;
-        xpointer<Task> m_pRootTask;
-        xpointer<Task> m_pRootGTask;
-        MARATHON_INSERT_PADDING(8);
-        xpointer<Task> m_pDocModeExecutor;
-        MARATHON_INSERT_PADDING(0x3C);
+        xpointer<RootTask> m_pRootTask;
+        xpointer<RootGTask> m_pRootGTask;
+        xpointer<ApplicationXenon> m_pDocCurrentApplication;
```

Comment:
Add a template function for casting to `ApplicationXenon`, this is an abstraction. Also missing an offset assertion for this field.

```suggestion
        xpointer<Application> m_pApplication;
```